packager/media/formats/mp4/mp4_muxer.cc

```diff
@@ -248,7 +248,10 @@ Status MP4Muxer::DelayInitializeMuxer() {
       ftyp->compatible_brands.push_back(FOURCC_cmfc);
 
     // Carry over movie extends header duration from init segment.
-    moov->extends.header.fragment_duration = streams()[0].get()->duration();
+    if (streams()[0].get()->duration() > 0 &&
+        streams()[0].get()->is_mehd_header_duration()) {
+      moov->extends.header.fragment_duration = streams()[0].get()->duration();
```

Comment:
Can we call `StreamInfo::get_default_fragment_duration()` here and only set the output if it's != 0?